### PR TITLE
Update DG on IS and AS

### LINKED ISSUE #214: Update `InterviewStatus` in DG and UG

### LINKED ISSUE #215: Update `ApplicationStatus` in DG and UG

### LINKED ISSUE #216: Update `Focus` Command on DG and UG

### PR DESCRIPTION
**Pull request by tiewweijian**
- Updated the Developers Guide for `InterviewStatus` and `ApplicationStatus`, and `Focus` 
- Removed the `InterviewStatus` trigger from the `EditCommand` segment. 
- Updated the use cases in the DG 

closes #214 
closes #215 
closes #216

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -271,33 +271,95 @@ This feature has been enhanced based on the initial implementation of the `edit`
 
 #### How is this feature implemented?
 
-The proposed edit mechanism is facilitated by `EditCommand`. It is supported by `EditCommandParser` where it will retrieve the attributes that can be edited.
+The proposed edit mechanism is facilitated by `EditCommand`. It is supported by `EditCommandParser` and `EditPersonDescriptor` 
+where it will retrieve the attributes that can be edited.
```
Comment (domlimm) [inline docs/DeveloperGuide.md:275]: Any reason why the sentence is cut to the next line? 😛

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -271,33 +271,95 @@ This feature has been enhanced based on the initial implementation of the `edit`
...
 * ApplicationStatus
-* InterviewStatus
 
 Based on the fields that are edited by the user, the EditCommandParser will retreive the information, and update the attributes accordingly.
-There are also some attributes that are dependent on another attribute, and we will introduce `Triggers` to update the attributes automatically.
-One of such is the `InterviewStatus`. It makes sense that if the application is accepted or rejected, it also means that the interview has been completed.
-Hence the `ApplicationStatus` will trigger `InterviewStatus` if the user is changing the `ApplicationStatus` from `Pending` to `Accepted` or `Rejected`.
-The `trigger mechanism` (WIP) is currently handled by `EditCommand#triggerInterviewStatus()`.
+This `EditPersonDescriptor` class will take in the current attributes of the Candidate, and update the fields accordingly by the
+`EditCommandParser`.
```
Comment (domlimm) [inline docs/DeveloperGuide.md:283]: This too. Just curious!

**Review by domlimm — APPROVED** [review]
All LGTM wj! Hopefully the resolution of this image `docs/images/ApplicationStatusCommandDiagram.png` comes out nicely. Nice job! 💯

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -271,33 +271,95 @@ This feature has been enhanced based on the initial implementation of the `edit`
...
+**1. We have implemented the `EditCommand` in this manner because it will help enhance the user experience. 
+One alternative method we had was to allow the user to update the candidate by retyping all the fields of the Candidate. 
+However, this proved to be not efficient, and hence we implemented it this way. With `EditCommandParser` and `EditPersonDescriptor`, 
+the user can edit one attribute of the candidate without typing every attribute of the Candidate.  
 
 #### UML Diagram
-The following activity diagram summarizes what happens when a user executes a `edit` command for `ApplicationStatus` and `InterviewStatus` <br>
-<img src="images/StatusActivityDiagram.png" width="250" />
+The following activity diagram summarizes what happens when a user executes a `edit` command. <br>
+<img src="images/EditCommandDiagram.png" width="250" />
```
Comment (lzan98) [inline docs/DeveloperGuide.md:294]: Might want to consider increasing the width to 550! Looks quite small now at 250!

**Review by lzan98 — COMMENTED** [review]
Hi @tiewweijian! Might want to consider resizing the `EditCommandDIagram`!

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (tiewweijian) [reply docs/DeveloperGuide.md:294]: ok sure will do !!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (tiewweijian) [reply docs/DeveloperGuide.md:283]: eh my bad ! I think i accidentally pushed to the next line on intellij!

**Review by lzan98 — APPROVED** [review]
LGTM! Thanks @tiewweijian !

**Pull request merged**